### PR TITLE
Support for testing NodeJS 16 in CentOS Stream 9

### PR DESCRIPTION
**Pull request by phracek**
This pull request adds support for building and testing NodeJS-16 in CentOS Stream 9.

The latest container-common-scripts were updated.
CentOS Stream 9 is not tested and built for versions, 12, 14, 14-minimal and 16-minimal.
Only version 16 supports CentOS Stream 9.

### CODE REVIEW

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
CentOS Stream 9 is failing for this reason:

```bash
STEP 6/11: RUN MODULE_DEPS="make gcc gcc-c++ libatomic_ops git openssl-devel" &&     INSTALL_PKGS="$MODULE_DEPS nodejs npm nodejs-nodemon nss_wrapper" &&     ln -s /usr/lib/node_modules/nodemon/bin/nodemon.js /usr/bin/nodemon &&     yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS &&     rpm -V $INSTALL_PKGS &&     yum -y clean all --enablerepo='*'
CentOS Stream 9 - BaseOS                        6.4 MB/s | 5.2 MB     00:00    
CentOS Stream 9 - AppStream                      14 MB/s |  10 MB     00:00    
CentOS Stream 9 - Extras packages               240  B/s | 257  B     00:01    
Last metadata expiration check: 0:00:01 ago on Mon Jan 24 09:57:02 2022.
No match for argument: libatomic_ops
Error: Unable to find a match: libatomic_ops
Error: error building at STEP "RUN MODULE_DEPS="make gcc gcc-c++ libatomic_ops git openssl-devel" &&     INSTALL_PKGS="$MODULE_DEPS nodejs npm nodejs-nodemon nss_wrapper" &&     ln -s /usr/lib/node_modules/nodemon/bin/nodemon.js /usr/bin/nodemon &&     yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS &&     rpm -V $INSTALL_PKGS &&     yum -y clean all --enablerepo='*'": error while running runtime: exit status 1
make[1]: Leaving directory '/tmp/sclorg/s2i-nodejs-container'
```

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
It looks like a some short Testing Farm outage

[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test-openshift]

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
Fedora test failed:

```bash
15:27:05                 out:  [PASSED] for 'clients' express
15:27:05                 out:  [FAILED] for 'clients' pino
15:27:05                 out:  [PASSED] for 'clients' prom-client
```

File: test/test-lib-nodejs.sh [hunk 1]
```diff
@@ -446,8 +446,13 @@ function test_nodemon_present() {
 
 function test_npm_cache_cleared() {
   # Test that the npm cache has been cleared
-  devdep=$(docker run --rm ${IMAGE_NAME}-testapp /bin/bash -c "! test -d \$(npm config get cache)")
-  check_result "$?"
+  # Do not run test on CentOS Stream 9.
```
Comment (zmiklank) [inline test/test-lib-nodejs.sh:449]: Why not run tests for c9s?

File: 16/Dockerfile.c9s [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    NPM_RUN=start \
+    NAME=nodejs \
+    NPM_CONFIG_PREFIX=$HOME/.npm-global \
+    PATH=$HOME/node_modules/.bin/:$HOME/.npm-global/bin/:$PATH \
+    CNB_STACK_ID=com.redhat.stacks.c9s-nodejs-16 \
+    CNB_USER_ID=1001 \
+    CNB_GROUP_ID=0
+
+ENV SUMMARY="Platform for building and running Node.js $NODEJS_VERSION applications" \
+    DESCRIPTION="Node.js $NODEJS_VERSION available as container is a base platform for \
```
[the remark this answers is not in the document]
Comment (pkubatrh) [reply 16/Dockerfile.c9s:29]: I would keep it as is, the tab is used to visually separate the env names from the ENV command

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-openshift]

**Comment by phracek** [comment]
[test]

File: test/test-lib-nodejs.sh (hunk 1 above)
Comment (pkubatrh) [reply test/test-lib-nodejs.sh:449]: Ran into this when checking RHEL9 test failures. This is actually caused by the npm command creating a `.npm/_logs` directory every time npm is called. Skipping this test is actually masking an issue with our assemble script as the cleanup cache step is not the last step in that script (hence we ship the `_logs` directory as well).
I will follow up with a proper fix once I think of a way to work around it.

**Pull request merged**